README.md

```diff
@@ -2,31 +2,100 @@
-re-purposing of different waste streams. A strong technical backbone, design and usability are key to the success of the
-platform!
+At Seenons, we are building a platform to support a circular economy. This encompasses collection, transportation, and re-purposing of different waste streams. The assignment you're about to undertake directly addresses two core aspects of our platform, which are integral to our mission:
+
+- **Finding service providers for waste collection**:
+  Enabling our customers to locate service providers who can collect their waste on a specified date.
+- **Registering a waste pickup**:
+  Allowing our customers to schedule a waste pickup with a service provider.
+
+These business cases form the basis of the use cases you will be implementing in this assignment.
 
 ## The assignment
 
 ---
 
-We have divided the assignment into what we called "gates", these gates are references to engineering skills that we are looking for in a future colleague.
-Some of them are required, some of them give you bonus points. You can read more about it below.
+### General Requirements
+
+For both use cases, certain validation rules apply universally:
+
+- Postal codes: Must be within the 0000 to 9999 range.
+- The application does not need to consider timezones for simplicity.
+- Waste streams: Must correspond to a predefined record in list of valid streams.
+- Service provider: Must corresponde to a prefedined record in a list of service providers.
+
+### Specific Use Case Requirements
+
+1. **Service Partners Availability**
+   Ensure the date for waste pickup is not before tomorrow, accounting for a next-day service policy.
+   see: <https://github.com/seenons/seenons-be-assignment/blob/main/src/usecases/availability/find_availability_at.ts>
 
-    The task is to at least complete two services
-    - Retrieve availability of service providers for a given location and date.
-    - Register the collection of a waste stream to be performed by a service provider.
+2. **Register Waste Pickup**
+   The selected date must be within the service provider's scheduling constaints and waste stream availability.
+   see: <https://github.com/seenons/seenons-be-assignment/blob/main/src/usecases/pickup/register_waste_pickup.ts>
 
-We purposely made this project framework-agnostic, and it only contains the bare minimum to perform the necessary use cases.
+Notes:
 
-## The output
+- You can use the provided code as a starting point or rebuild it however you see fit, including the domain representations in case you do not agree with its current contents.
+
+## The delivery
+
+---
+
+- Create a **private** copy of this repository. Do not fork it.
+- Create a new readme file that explains your take on the assignment, your thought process, and any other information that you believe is relevant for the review.
+- Implement the necessary changes to at least achieve the requirements.
+- Invite <caio@seenons.com> as a collaborator to your private repository.
```

Comment:
I don't know if Github allows this, but can this be `@seenons/backenders` instead? You do the reviews now, but that might not always be the case. It's better if everybody has access.

Reply:
Sadly no, we could use a bot (perhaps wall-e for that)